File: tasks/Two_Sum/Python3/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        list_map = dict()
        for i in range(len(nums)):
            if list_map.get(target - nums[i]) is None:
                list_map[nums[i]] = i
            else:
                return [list_map[target - nums[i]], i]

File: tasks/Two_Sum/Cpp/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> numToInd;
        vector<int> result;
        for (int i = 0; i < nums.size(); i++) {
            const auto it = numToInd.find(target - nums[i]);
            if (it == numToInd.end()) {
                numToInd[nums[i]] = i;
            } else {
                result = {it->second, i};
                break;
            }
        }
        return result;
    }
};